go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -1142,12 +1142,12 @@ func getPodNamespacedName(pod *corev1.Pod) ktypes.NamespacedName {
 	return ktypes.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}
 }
 
-func generateIPTablesSNATRuleArg(srcIP net.IP, infName, snatIP string) iptables.RuleArg {
+func generateIPTablesSNATRuleArg(srcIP net.IP, isIPv6 bool, infName, snatIP string) iptables.RuleArg {
 	var srcIPFullMask string
-	if srcIP.To4() != nil { // v4
-		srcIPFullMask = fmt.Sprintf("%s/32", srcIP.String())
-	} else { // v6
+	if isIPv6 {
 		srcIPFullMask = fmt.Sprintf("%s/128", srcIP.String())
+	} else {
+		srcIPFullMask = fmt.Sprintf("%s/32", srcIP.String())
```

Comment:
NIT: I thought that iptables is smart enough to take an address w/out mask and know that it is a host, no?
but that is okay to explicitly provide the host mask, ofc.

Reply:
Interesting. Ill look into this in the future and adjust it. I wont do it in this PR.

Reply:
I have tested with this method and I dont want to make the changes now without testing it and especially in this PR.